test/battle/CHealthTest.cpp

```diff
@@ -86,155 +77,167 @@ static void checkHeal(CHealth & health, EHealLevel level, EHealPower power, cons
-	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT - 1);
-	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+	EXPECT_EQ(health.getCount(), UNIT_AMOUNT - 1);
+	EXPECT_EQ(health.getFirstHPleft(), UNIT_HEALTH);
+	EXPECT_EQ(health.getResurrected(), 0);
 
 	health.init();
 
 	checkNormalDamage(health, UNIT_HEALTH * UNIT_AMOUNT);
-	checkEmptyHealth(health, *this);
+	checkEmptyHealth(health, mock);
 
 	checkHeal(health, EHealLevel::RESURRECT, EHealPower::ONE_BATTLE, UNIT_HEALTH * UNIT_AMOUNT, UNIT_HEALTH * UNIT_AMOUNT);
-	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
-	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getResurrected(), UNIT_AMOUNT);
+	EXPECT_EQ(health.getCount(), UNIT_AMOUNT);
+	EXPECT_EQ(health.getFirstHPleft(), UNIT_HEALTH);
+	EXPECT_EQ(health.getResurrected(), UNIT_AMOUNT);
 
 	health.takeResurrected();
-	checkEmptyHealth(health, *this);
+	checkEmptyHealth(health, mock);
 }
 
-BOOST_FIXTURE_TEST_CASE(resurrectPermanent, CUnitHealthInfoMock)
+TEST_F(HealthTest, resurrectPermanent)
 {
+	setDefaultExpectations();
+	health.init();
+
 	checkNormalDamage(health, UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT - 1);
-	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+	EXPECT_EQ(health.getCount(), UNIT_AMOUNT - 1);
+	EXPECT_EQ(health.getFirstHPleft(), UNIT_HEALTH);
+	EXPECT_EQ(health.getResurrected(), 0);
 
 	checkHeal(health, EHealLevel::RESURRECT, EHealPower::PERMANENT, UNIT_HEALTH, UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT);
-	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+	EXPECT_EQ(health.getCount(), UNIT_AMOUNT);
+	EXPECT_EQ(health.getFirstHPleft(), UNIT_HEALTH);
+	EXPECT_EQ(health.getResurrected(), 0);
 
 	checkNormalDamage(health, UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getCount(), UNIT_AMOUNT - 1);
-	BOOST_CHECK_EQUAL(health.getFirstHPleft(), UNIT_HEALTH);
-	BOOST_CHECK_EQUAL(health.getResurrected(), 0);
+	EXPECT_EQ(health.getCount(), UNIT_AMOUNT - 1);
+	EXPECT_EQ(health.getFirstHPleft(), UNIT_HEALTH);
+	EXPECT_EQ(health.getResurrected(), 0);
 
 	health.init();
 
 	checkNormalDamage(health, UNIT_HEALTH * UNIT_AMOUNT);
-	checkEmptyHealth(health, *this);
+	checkEmptyHealth(health, mock);
 
 	checkHeal(health, EHealLevel::RESURRECT, EHealPower::PERMANENT, UNIT_HEALTH * UNIT_AMOUNT, UNIT_HEALTH * UNIT_AMOUNT);
-	checkFullHealth(health, *this);
+	checkFullHealth(health, mock);
 
 	health.takeResurrected();
-	checkFullHealth(health, *this);
+	checkFullHealth(health, mock);
 }
 
-BOOST_FIXTURE_TEST_CASE(singleUnitStack, CUnitHealthInfoMock)
+TEST_F(HealthTest, singleUnitStack)
 {
 	//related to issue 2612
 
 	//one Titan
-	baseAmount = 1;
-	maxHealth = 300;
+
+	EXPECT_CALL(mock, unitMaxHealth()).WillRepeatedly(Return(1));
+	EXPECT_CALL(mock, unitBaseAmount()).WillRepeatedly(Return(300));
```

Comment:
unitMaxHealth should be 300 and unitBaseAmount should be 1.
You can leave unitMaxHealth = 1, unitBaseAmount = 300 as separate test case, it also should pass.